ampersand-calendar.js

```diff
@@ -0,0 +1,290 @@
+    initialize: function() {
+      this.currentDates = this.createDateArray();
+    },
+    selectPreviousMonth: function() {
+      this.current.subtract(1, 'M');
+      this.currentDates = this.createDateArray();
+    },
+    selectNextMonth: function() {
+      if (this.current.year() < moment().year() || (this.current.year() === moment().year() && this.current.month() < moment().month())) {
+        this.current.add(1, 'M');
+        this.currentDates = this.createDateArray();
+      }
+    },
+    selectWeekday: function(event) {
+      d3.select(event.target).each(function(d) {
+        if (d.valid) {
+          var date = d.date;
+          var now = moment();
+          if (date.diff(now, 'days') < 0 || date.isSame(now, 'day')) {
+            if (this.startDate === null || this.endDate !== null) {
+              this.startDate = date;
+              this.endDate = null;
+            } else {
+              var diff = date.diff(this.startDate, 'days');
+              if (diff === 0) {
+                this.startDate = null;
+              } else if (diff < 0) {
+                this.endDate = this.startDate;
+                this.startDate = date;
+              } else {
+                this.endDate = date;
+              }
+            }
+            this.currentDates = this.createDateArray();
+          }
+        }
+      }.bind(this));
+    },
+    createDateArray: function() {
+      var month = moment(this.current);
+
+      var startDayOfMonth = month.date(1).day();
+      var lastDateOfMonth = month.add(1, 'M').date(0).date();
+
+      var dates = [];
+      var now = moment();
+      var weekday;
+      var week;
+      var data;
+      var selected;
+      var dateMoment;
+      for (var i = 0; i < startDayOfMonth + lastDateOfMonth; i++) {
+        var date = i - startDayOfMonth + 1;
+        dateMoment = moment(month).date(date);
+        weekday = i % 7;
+        week = Math.floor(i / 7);
+
+        var valid = false;
+        selected = false;
+
+        valid = i >= startDayOfMonth && (dateMoment.diff(now, 'days') < 0 || dateMoment.isSame(now, 'day'));
+
+        if (this.endDate !== null) {
+          selected = (this.startDate !== null && dateMoment.diff(this.startDate, 'days') >= 0) && dateMoment.diff(this.endDate, 'days') <= 0;
+        } else {
+          selected = (this.startDate !== null && dateMoment.diff(this.startDate, 'days') === 0);
+        }
+
+        data = {
+          valid: valid,
+          date: dateMoment,
+          selected: selected,
+          weekday: weekday,
+          week: week
+        };
+
+        dates.push(data);
+      }
+
+      weekday ++;
+      month.add(1, 'M');
+      for (var j = 0; j < (7 * (6 - week)) - weekday; j++) {
+        dateMoment = moment(month).date(j + 1);
+        selected = true;
+
+        if (this.endDate !== null) {
+          selected = (this.startDate !== null && dateMoment.diff(this.startDate, 'days') >= 0) && dateMoment.diff(this.endDate, 'days') <= 0;
```

Comment:
did you try using things like moment.isSame and moment.isAfter?  http://momentjs.com/docs/#/query/